client.go

```diff
@@ -145,50 +145,56 @@ func (c *Client) registerRPCProvider() error {
 	return nil
 }
 
-func (c *Client) registerProviders() {
-	// register the rpc provider
-	// without the consumer URL there is no way to send RPC requests.
-	if c.providerConfig.rpc.ConsumerURL != "" {
-		// when the rpc provider is to be used, we won't register any other providers.
-		err := c.registerRPCProvider()
-		if err == nil {
-			c.Logger.Info("note: with the rpc provider registered, no other providers will be registered and available")
-			return
-		}
-		c.Logger.Info("failed to register rpc provider, falling back to registering all other providers", "error", err.Error())
-	}
-	// register ipmitool provider
+// register ipmitool provider
+func (c *Client) registerIPMIProvider() error {
 	ipmiOpts := []ipmitool.Option{
 		ipmitool.WithLogger(c.Logger),
 		ipmitool.WithPort(c.providerConfig.ipmitool.Port),
 		ipmitool.WithCipherSuite(c.providerConfig.ipmitool.CipherSuite),
 		ipmitool.WithIpmitoolPath(c.providerConfig.ipmitool.IpmitoolPath),
 	}
-	if driverIpmitool, err := ipmitool.New(c.Auth.Host, c.Auth.User, c.Auth.Pass, ipmiOpts...); err == nil {
-		c.Registry.Register(ipmitool.ProviderName, ipmitool.ProviderProtocol, ipmitool.Features, nil, driverIpmitool)
-	} else {
-		c.Logger.Info("ipmitool provider not available", "error", err.Error())
+
+	driverIpmitool, err := ipmitool.New(c.Auth.Host, c.Auth.User, c.Auth.Pass, ipmiOpts...)
+	if err != nil {
+		return err
 	}
 
-	// register ASRR vendorapi provider
+	c.Registry.Register(ipmitool.ProviderName, ipmitool.ProviderProtocol, ipmitool.Features, nil, driverIpmitool)
+
+	return nil
+}
+
+// register ASRR vendorapi provider
+func (c *Client) registerASRRProvider() error {
```

Comment:
this func will never return an error. i would suggest not having any return value.

Reply:
That was mainly keeping the pattern from the above function since I was looking into if these could be turned into an interface method which providers implement to register themselves so this can be moved into each provider.

I can remove the extra error returns for now.